From a6edc818e8bdafc31cb7b7fe1f526c325561bb5f Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Tue, 6 Jun 2017 16:32:52 +0800
Subject: [PATCH 0903/1200] dts: a3700: add topaz switch support for
 espressobin

Espressobin board has a SOHO switch which is connected in multi-chip
addressing mode with phy address of 1. The switch has 4 ports, port 0
is connected to neta0 and forced link-up.

Change-Id: I461b48fac19eca3ce8a211863f1023c166f97350
Signed-off-by: Ken Ma <make@marvell.com>
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/60944
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
---
 arch/arm/dts/armada-3720-espressobin.dts | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm/dts/armada-3720-espressobin.dts b/arch/arm/dts/armada-3720-espressobin.dts
index 53e96889e0..d8bd75211b 100644
--- a/arch/arm/dts/armada-3720-espressobin.dts
+++ b/arch/arm/dts/armada-3720-espressobin.dts
@@ -67,6 +67,20 @@
 		device_type = "memory";
 		reg = <0x00000000 0x00000000 0x00000000 0x20000000>;
 	};
+
+	mdio@32004 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		status = "okay";
+
+		mvswitch: switch@000000{
+			compatible = "marvell,mv88e6xxx";
+			status = "okay";
+			phy-addr = <1>;
+			cpu-port = <0x0>;
+			port-mask = <0xf>;
+		};
+	};
 };
 
 &comphy {
@@ -93,6 +107,8 @@
 	pinctrl-0 = <&rgmii_pins>, <&smi_pins>;
 	phy-mode = "rgmii";
 	phy_addr = <0x1>;
+	phy = <&mvswitch>;
+
 	fixed-link {
 		speed = <1000>;
 		full-duplex;
-- 
2.22.0

